app/scripts/modules/core/src/managed/overview/EnvironmentsOverview.tsx

```diff
@@ -63,9 +63,11 @@ const EnvironmentOverview = ({ appName, environment }: IEnvironmentProps) => {
   return (
     <BaseEnvironment title={environment.name}>
       <CollapsibleSection heading="Artifacts" {...sectionProps} defaultExpanded enableCaching={false}>
-        {state.artifacts?.map((artifact) => (
-          <Artifact key={artifact.reference} artifact={artifact} />
-        ))}
+        {state.artifacts ? (
```

Comment:
Should this also check `artifacts.length`?

Reply:
Yeah, it won't hurt :)